### ISSUE TITLE
Crash on opening a window in Gnome 3.24.1

### ISSUE THREAD

**Issue by evaryont** [timbertson/shellshape#190]
I'm not sure how to debug this, but gnome-shell seems to crash when I open a new window in a fresh session.

When I enable shellshape during a session, it seems to work as expected. But if I log out and then log in, the first window I open immediately closes the session, putting back at the login screen (gdm).

Looking through `journalctl --user`, I can only guess that these log lines are related:

```
May 13 13:43:06 gazelle.evaryont.me gnome-shell[8465]: JS WARNING: [/home/colin/.local/share/gnome-shell/extensions/shellshape@gfxmonk.net/extension.js 2097]: reference to undefined property self.extension.focus_window
May 13 13:43:06 gazelle.evaryont.me org.gnome.Shell.desktop[8465]: WARN : Layout.tile_for(null)
```

Running in Archlinux.

**Comment by evaryont** [comment]
I noticed I had some updates pending, and updated to 3.24.2. It still crashes.

Here is the full log, including installation. I open Epiphany, install the extension (after deleting it with gnome-tweak-tool), close epiphany, and then attempt to open a terminal. So I don't even have to log out first to cause the crash.

```
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]:   inflating: /home/colin/.local/share/gnome-shell/extensions/shellshape@gfxmonk.net/prefs.js
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]:   inflating: /home/colin/.local/share/gnome-shell/extensions/shellshape@gfxmonk.net/data/glib-2.0/schemas/g
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]:   inflating: /home/colin/.local/share/gnome-shell/extensions/shellshape@gfxmonk.net/extension.js
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]:  extracting: /home/colin/.local/share/gnome-shell/extensions/shellshape@gfxmonk.net/metadata.json
May 13 14:22:55 gazelle.evaryont.me gnome-shell[2014]: [Dynamic Panel Transparency] User theme changed.
May 13 14:22:55 gazelle.evaryont.me gnome-shell[2014]: adding icon dir: /home/colin/.local/share/gnome-shell/extensions/shellshape@gfxmonk.net/data/icons
May 13 14:22:55 gazelle.evaryont.me gnome-shell[2014]: JS WARNING: [/home/colin/.local/share/gnome-shell/extensions/shellshape@gfxmonk.net/extension.js 1563]: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: INFO : initting schemas
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: WARN : loading schema from: /home/colin/.local/share/gnome-shell/extensions/shellshape@gfxmonk.net/data/glib-2.0/schemas
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: INFO : translations initted for shellshape
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: INFO : shellshape initialized!
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: INFO : shellshape enable() called
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: INFO : initting schemas
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: WARN : loading schema from: /home/colin/.local/share/gnome-shell/extensions/shellshape@gfxmonk.net/data/glib-2.0/schemas
May 13 14:22:55 gazelle.evaryont.me gnome-shell[2014]: JS WARNING: [/home/colin/.local/share/gnome-shell/extensions/shellshape@gfxmonk.net/extension.js 2786]: reference to undefined property Shell[KeyBindingMode].MESSAGE_TRAY
May 13 14:22:55 gazelle.evaryont.me gnome-shell[2014]: JS WARNING: [/home/colin/.local/share/gnome-shell/extensions/shellshape@gfxmonk.net/extension.js 2097]: reference to undefined property self.extension.focus_window
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: WARN : Layout.tile_for(null)
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: WARN : Layout.tile_for(null)
May 13 14:22:55 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: INFO : shellshape enabled
May 13 14:22:55 gazelle.evaryont.me gnome-shell[2014]: [Dynamic Panel Transparency] No background image found in user theme.
May 13 14:22:55 gazelle.evaryont.me gnome-shell[2014]: [Dynamic Panel Transparency] No border image found in user theme.
May 13 14:22:55 gazelle.evaryont.me gnome-shell[2014]: [Dynamic Panel Transparency] Detected user theme style: rgba(0, 0, 0, 255)
May 13 14:22:58 gazelle.evaryont.me org.gnome.Epiphany.desktop[4731]: Error sending IPC message: Broken pipe
May 13 14:23:00 gazelle.evaryont.me org.gnome.SettingsDaemon.MediaKeys.desktop[2161]: Warning: DESKTOP_STARTUP_ID not set and no fallback available.
May 13 14:23:00 gazelle.evaryont.me dbus-daemon[1980]: Activating via systemd: service name='org.gnome.Terminal' unit='gnome-terminal-server.service'
May 13 14:23:00 gazelle.evaryont.me systemd[1963]: Starting GNOME Terminal Server...
May 13 14:23:01 gazelle.evaryont.me dbus-daemon[1980]: Successfully activated service 'org.gnome.Terminal'
May 13 14:23:01 gazelle.evaryont.me systemd[1963]: Started GNOME Terminal Server.
May 13 14:23:01 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: WARN : Layout.tile_for(null)
May 13 14:23:01 gazelle.evaryont.me gnome-terminal-[5292]: Error reading events from display: Broken pipe
May 13 14:23:01 gazelle.evaryont.me gnome-session[1982]: gnome-session-binary[1982]: WARNING: Application 'org.gnome.Shell.desktop' killed by signal 11
May 13 14:23:01 gazelle.evaryont.me pulseaudio[2064]: ICE default IO error handler doing an exit(), pid = 2064, errno = 11
May 13 14:23:01 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: (EE)
May 13 14:23:01 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: Fatal server error:
May 13 14:23:01 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: (EE) failed to read Wayland events: Broken pipe
May 13 14:23:01 gazelle.evaryont.me org.gnome.Shell.desktop[2014]: (EE)
May 13 14:23:01 gazelle.evaryont.me evolution-alarm[2351]: Error reading events from display: Broken pipe
May 13 14:23:01 gazelle.evaryont.me unknown[4559]: Error reading events from display: Broken pipe
May 13 14:23:01 gazelle.evaryont.me gnome-session-binary[1982]: WARNING: Application 'org.gnome.Shell.desktop' killed by signal 11
May 13 14:23:01 gazelle.evaryont.me gnome-session-binary[1982]: Unrecoverable failure in required component org.gnome.Shell.desktop
```

**Comment by timbertson** [comment]
There's definitely a bug in gnome-shell or one of its native components here, since JS extensions like shellshape can't themselves cause a segfault (signal 11). Nothing in this log points to an obvious shellshape bug to me, so I don't know that there's much I can do. It looks like you may be running wayland; you could try running with X instead?

**Comment by evaryont** [comment]
gnome-shell v3.24.3 on X11 does indeed work. I had another segfault when I first enabled the extension. However, after logging back in from GDM it seems to be working well. I'll try it again within Wayland, see if some update between then and now as improved the situation.

**Comment by timbertson** [comment]
Closing as duplicate of (fixed) #169
